### PR TITLE
Disable update button when no authentication methods selected

### PR DESCRIPTION
**Pull request by Rashmini**
### Purpose
"Update" button is disabled when there is only one authentication step and there are no authentication options selected in that step.

### Related Issues
- [https://github.com/wso2/product-is/issues/11854](https://github.com/wso2/product-is/issues/11854)

### Checklist
- [x] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation
- [ ] Documentation provided. (Add links)
- [ ] Unit tests provided. (Add links if any)
- [ ] Integration tests provided. (Add links if any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx [hunk 1]
```diff
@@ -242,6 +247,20 @@ export const StepBasedFlow: FunctionComponent<AuthenticationFlowPropsInterface>
         }
     }, [ authenticationSteps ]);
 
+    /**
+     * Change button disable state when authentication Steps are changed.
+     */
+    useEffect(() => {
+
+        const noOptionsSelected = authenticationSteps.length === 1 && authenticationSteps[ 0 ].options?.length === 0;
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx:255]:
```suggestion
        const noOptionsSelected: boolean = authenticationSteps.length === 1 && authenticationSteps[ 0 ].options?.length === 0;
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx [hunk 2]
```diff
@@ -96,6 +96,10 @@ interface AuthenticationFlowPropsInterface extends TestableComponentInterface {
      * Update authentication steps.
      */
     updateSteps: (add: boolean) => void;
+    /**
+     * Callback to update the button disable state change.
+     */
+    onAuthenticationSequenceChange: (buttonStateDisabled: boolean) => void;
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx:102]: Shall we rename this to `isDisabled`? WDYT?
```suggestion
    onAuthenticationSequenceChange: (isDisabled: boolean) => void;
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx (hunk 1 above)
Comment (Rashmini) [reply apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx]: fixed in 7402005b3cafad3dd06a3f785122cb2d30e11cf4

Comment (Rashmini) [reply apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx]: fixed in 7402005b3cafad3dd06a3f785122cb2d30e11cf4

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx (hunk 2 above)
Comment (Rashmini) [reply apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx]: fixed in 7402005b3cafad3dd06a3f785122cb2d30e11cf4

**Review by brionmario — APPROVED** [review]

**Pull request merged**